### PR TITLE
Fix graph iterate vertices/edges in page error

### LINKED ISSUE #13: graph.iterateEdges(1000) throw exception
异常信息：
``` 
Not support quering edges based on paging and [vertex, direction, label, properties, offset] together 
```

[详细](https://github.com/hugegraph/hugegraph/issues/158)

BUG 原因有两个：
1、目前服务端的分页获取API不支持传入`label、properties`等参数，但是客户端的`GraphAPI`类的`formatProperties()`将本来是`null`的`properties`转成了空`map`然后序列化为`"{}"`，导致服务端接受的`properties`不为`null`。
2、客户端的分页迭代器没有考虑好服务端已经没有下一页的情况。

### PR DESCRIPTION
**Pull request by Linary**
Fix #13

Change-Id: I98371718af0b2f73ec687f4ba3eb5a42ae4954ea

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/structure/graph/GraphIterator.java [hunk 1]
```diff
@@ -31,33 +31,45 @@
...
 
     @Override
     public boolean hasNext() {
-        if (this.results == null || this.cursor >= this.results.size()) {
-            Pageable<T> pageable = this.pageFetcher.apply(this.page);
-            this.results = pageable.results();
-            this.page = pageable.page();
-            this.cursor = 0;
+        if ((this.results == null || this.cursor >= this.results.size()) &&
+            !this.finished) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/graph/GraphIterator.java]: move "!this.finished" into fetch()

File: src/main/java/com/baidu/hugegraph/structure/graph/GraphIterator.java [hunk 2]
```diff
@@ -31,33 +31,45 @@
...
+            this.fetch();
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/graph/GraphIterator.java]: assert this.results != null

File: src/test/java/com/baidu/hugegraph/functional/FuncTestSuite.java [hunk 3]
```diff
@@ -24,11 +24,12 @@
...
-    PropertyKeyTest.class,
-    VertexLabelTest.class,
-    EdgeLabelTest.class,
-    VertexTest.class,
-    EdgeTest.class
+//    PropertyKeyTest.class,
+//    VertexLabelTest.class,
+//    EdgeLabelTest.class,
+//    VertexTest.class,
+//    EdgeTest.class,
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/functional/FuncTestSuite.java]: uncomment

File: src/main/java/com/baidu/hugegraph/driver/GraphManager.java [hunk 4]
```diff
@@ -128,14 +129,16 @@ public Vertex getVertex(Object vertexId) {
         return this.iterateVertices(null, null, sizePerPage);
     }
 
+    @UnimplementedFeature(desc = "label must be null")
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/GraphManager.java]: Server does not support query label by page

**Pull request merged**